### DISCUSSION TITLE
KTL 18.8 issues

### DISCUSSION THREAD

**Discussion by jbk68** [cortexrd/Knack-Toolkit-Library#154] [General]
Prior to update and primarily on mobile app access, I had a menu drop down ("hamburger" looking menu) on the upper left hand corner next to our company logo.  Now with 18.8, we still have left hand corner menu but now have the name of the person logged in as well as links to pages they have access to right below the company logo.  **NO CHANGES TO SETTINGS** were made, just a simple refresh to app to update to 18.8.  Page now looks terrible with the user name and links being shown.  I checked all the KTL settings and none have them have changed.

Just a fyi, we have (3) separate apps that we have developed with similar layouts and all (3) have 18.8 and have the same annoying issues as above.
 
Any and all help is appreciated... JON

**Comment by cortexrd** [comment]
Hi Jon,

I'd like to see what you're referring to.  I tried my tutorial app on my cell phone and it looks like this:

![image](https://github.com/cortexrd/Knack-Toolkit-Library/assets/42851359/c0a1069d-83e9-45ee-9e86-5777b4ba68e3)

In the green rectangle, is this what you're referring to?
Can you provide a screen grab of your cell phone?

All the recent changes don't seem to be related to that.  Even in the several last releases.
If you remove the KTL, all goes away and looks fine?

And if you go back one or two versions, what happens?
For example, you can safely try this to go back to last release **0.18.7**:

```
KnackInitAsync = function ($, callback) { // Load the Knack Toolkit Library (KTL)
    (window.LazyLoad = LazyLoad) && LazyLoad.js(['https://ctrnd.s3.amazonaws.com/Lib/KTL/KTL_Start.js'], () => {
        loadKtl($, callback, (typeof KnackApp === 'function' ? KnackApp : null), '0.18.7' /*KTL version, leave blank to get latest*/, 'full'/*min or full*/);
    })
};
```


To test any other previous version, you can see all valid numbers here:

https://github.com/cortexrd/Knack-Toolkit-Library/tags

I'd need a bit more information.
If you could create an account for me using nd@ctrnd.com it would be ideal.  Just a dummy page with nothing on it, as long as I can see your bug.  Send me the password in a private email.

Norm

**Reply by jbk68** [reply]
Yes, what you see in the green is exactly what I now see.  Before 18.8 this wasn't present.

**Comment by cortexrd** [comment]
Any additional information, as asked?

**Reply by jbk68** [reply]
Just got in front of a computer.  I tried changing version back to 18.7 and earlier even versions with no change.  Still seeing the something similar to what your seeing in green rectangle above starting with logged in as.  I'm more than certain this wasn't there just a couple days ago and I've made no changes to scripts on any of the (3) apps its effecting.

What more do you need from me ???

**Reply by cortexrd** [reply]
Jon,

I believe you!

But I also think that what you see is not related to the KTL.

BTW... did you try removing the KTL code completely as asked?  That would prove my hypothesis.
Here, after removing all Javascript code from my Tutorial app, I see exactly the same thing as in my screen grab - except for the keywords that are now showing up as expected.

That's proof that KTL is not involved.  If you go back to an older KTL version and you still have this, it's not the KTL.

Now, how can we address this?  A new keyword?  New parameter?

Norm

**Reply by jbk68** [reply]
I completely removed KTL from one of the apps and the issue still was there and as you said, proofs that its not a KTL issue.

Just weird that it just happened the last couple days.  I tried going into one of apps Knack "Live Design" settings and no matter what I did, still had this annoying logged in as and page links showing right below app name / logo.

I appreciate your help Norm and hope you find a solution to this issue... JON

**Reply by jbk68** [reply]
Norm, placing the following CSS script removed the annoying "logged in as" and page links:

@media only screen and (max-width: 767px) {
  /* Hide kn-info-bar on mobile */
  .kn-info-bar {
    display: none !important;
  }
}

**Comment by CSWinnall** [comment]
Hi Jon

You are right

That information used to be in the hamburger menu.  I guess this is an update to knack.  Thanks for spotting this as we don't predominantly use mobile for our app and it does take up real estate.

Craig

**Comment by jbk68** [comment]
Hey Craig, for some reason the information appeared just below the header / logo area which as you said takes up real estate and looks horrible (see Norm's example above).  The information still was in the "hamburger menu" but for some reason also un-necessarily appeared also below header.  Once I put in the script above, it fixed the issue.  

I have reported this issue to Knack this morning and they are currently looking into it.

Thanks... JON

**Reply by CSWinnall** [reply]
Ok if they get back to you would you let us know what they say. And we will implement the CSS you posted although the only change I would suggest is making your selector more specific rather than using ! important 
.kn-content .kn-info-bar
Or 
#knack-body .kn-content .kn-info-bar

As the important should be used sparingly

Craig

**Reply by jbk68** [reply]
Will do...

**Comment by cortexrd** [comment]
Hi guys,

This is a good solution that only removes the info bar if the device is mobile.

Add this anywhere inside the ktlReady section.

```
ktl.core.waitSelector('.kn-info-bar').then(() => {
    if (Knack.isMobile())
        $('.kn-info-bar').remove();
})
```

Enjoy,
Norm
